src/framework/types.hpp

```diff
@@ -39,10 +39,16 @@ namespace AER {
   using int_t = int_fast64_t;
   using uint_t = uint_fast64_t;
   using complex_t = std::complex<double>;
+  using complexf_t = std::complex<float>;
```

Comment:
I'm hesitant to add more aliases here. Maybe instead look to change cvector and cmatrix to be templated as:
```
template <class FloatType = double>
using cmatrix_t = matrix<std::complex<FloatType>>
```
so we can use templates in all the other util functions to work for all float precisions (This would involve a pretty big refactor in a lot of places though)

Reply:
I found them helpful when I was writing the tests for double/float specialization... but yes, I do prefer your suggestion.

Reply:
You can always add them to the top of the test file

Reply:
As this would require a large refactor, what about  adding just the 3 single-precision types: `complexf_t`, `cvectorf_t`, `cmatrixf_t` ? And we can tackle this refactor later in another PR (to not overcomplicate this one).